### DISCUSSION TITLE
Features ideas

### DISCUSSION THREAD

**Discussion by snesrev** [snesrev/zelda3#62] [Ideas]
Talk about new features here

**Comment by Ruggie74** [comment]
I do not know ANYTHING about making games. Is it possible to redo ALL of the terrain and sprite graphics/animations, or do those features suffer similar restrictions as SNES romhacks? What about adding things like 3D light sources? Custom weapons/tools from more modern Zelda games being implemented? Someone with more knowledge please discuss what the possibilities are with this project because I don't have a damn clue what I am talking about!

**Reply by UltraHDR** [reply]
Sprite replacement would be cool

**Comment by UltraHDR** [comment]
MSU1 audio/video replacement support
https://www.youtube.com/watch?v=8AKSoDFUSKA

SNES audio samples can be found here https://docs.google.com/spreadsheets/d/1JJBlHHDc65fhZmKUGLrDTLCm6rfUU83-kbuD8Y0zU0o

**Reply by Ruggie74** [reply]
Yes please.

**Reply by snesrev** [reply]
It's added in master (for music)

**Reply by Ruggie74** [reply]
> It's added in master (for music)

You got MSU-1 support working for this?! OMG I CAN'T WAIT!! Thank you so much. Can anyone please share a video of what they have done with a MSU-1 patch? I would really like to hear some of the music people are adding to the game so that I can be inspired to try and learn how to make a MSU-1 patch myself!! Thank you guys!

**Comment by UltraHDR** [comment]
**Add BS The Legend of Zelda & BS The Legend of Zelda: Ancient Stone Tablets support**

Not sure if it uses the same engine

https://zelda.fandom.com/wiki/BS_The_Legend_of_Zelda
https://zelda.fandom.com/wiki/BS_The_Legend_of_Zelda:_Ancient_Stone_Tablets

Thanks

**Reply by Ruggie74** [reply]
Honestly though, do you think that just having sprite replacement would satisfy this idea? Or no?

**Reply by UltraHDR** [reply]
No, the first is a remake of the NES https://en.wikipedia.org/wiki/The_Legend_of_Zelda_(video_game) game and the 2nd is a completely different game

**Comment by UltraHDR** [comment]
Add features from "A Link to the Past Redux" & "A Link to the Past DX" romhacks

https://www.romhacking.net/hacks/2594/
https://www.romhacking.net/hacks/2234/

e.g: 

- Break pots with sword
- Collect items with sword

**Reply by Ruggie74** [reply]
Great ideas, especially the 2nd item slot from those hacks!

**Reply by JasonShearer** [reply]
Nothing against UltraHDR's suggestions, but I think they belong in a fork.

**Reply by UltraHDR** [reply]
Yes I agree it's probably suited for a fork

**Reply by snesrev** [reply]
Collect items with sword added in master
Break pots with sword added in master

**Comment by UltraHDR** [comment]
Use X Button for a 2nd item slot and remap Map to another button (maybe press select and start together)

**Reply by Ruggie74** [reply]
Hahah YES!!

**Reply by snesrev** [reply]
Added L/R for switching between items.

**Reply by snesrev** [reply]
X button for 2nd item slot added in master.

**Reply by FitzRoyX** [reply]
I have my doubts about messing with original bindings like this. It seems like something that could be triggered by accident and cause people to think their map isn't working.

**Reply by snesrev** [reply]
It has to be enabled in zelda3.ini though (even though ItemSwitchLR is enabled by default)... Any other ideas?

**Reply by CowSoftware** [reply]
I think adding a warning or disclaimer in the .ini file and informing the player about the map button getting rebound to another button would make things less confusing. Something like this:
![image](https://user-images.githubusercontent.com/82652849/191617028-577e2144-081a-497b-881e-cb96f190481c.png)

...Or you can make this feature togglable.

**Reply by FitzRoyX** [reply]
Oh, you tied it to the LR stuff. Yeah I dunno, maybe separate them. Although I question how many people would use a feature that makes map access more annoying or impossible. Was this in a popular hack or something?

**Reply by snesrev** [reply]
Is Select instead of X more annoying?

**Reply by FitzRoyX** [reply]
I thought select was used for the save/quit prompt?

**Reply by snesrev** [reply]
I moved the old Select menu to the Start menu. I.e. first go to Start, then press Select. And then Select is available to use for the map.

**Comment by Ruggie74** [comment]
Allow L and R buttons to cycle through your usable items/tools.

**Reply by snesrev** [reply]
Added in master

**Reply by FitzRoyX** [reply]
Pretty reasonable feature since they weren't bound to anything outside the mode7 map. Maybe shovel access can be tied to LR after flute acquisition so that you don't have to redesign the menu?

**Comment by Witchybun** [comment]
If there was effort to include randomizer support akin to Ship of Harkinian I think it would really streamline how people do randomizers; however I do wonder how that community will react to how fast some of the load times are when it comes to races.

**Comment by snesrev** [comment]
How do you like the inventory reordering?

**Comment by VictorXPDE** [comment]
Add support for translations and Unicode characters. The language could be changed in the .ini file.

**Comment by CowSoftware** [comment]
Add the ability to mute the low HP alarm sound.

**Reply by snesrev** [reply]
Added in master

**Comment by JasonShearer** [comment]
Is it possible to add optional skipping of the Triforce intro upon button press and yellow text when rupees/bombs/arrows are at maximum like in the GBA port? A whole list of differences between the SNES and GBA versions can be found [here](http://www.jaytheham.com/zcw/A_Link_to_the_Past_Version_Guide). Perhaps there's some other useful changes on there that can be incorporated as optional features.

**Reply by snesrev** [reply]
Triforce intro skipping added in master

**Reply by snesrev** [reply]
Max items can now be displayed with a yellow/brownish color.

**Comment by JasonShearer** [comment]
I already requested this under Issues, which is still marked as Open, but I might as well include it here.

Pixel shader/filter support. :D  [.cg(p)](https://github.com/libretro/common-shaders), [.glsl(p)](https://github.com/libretro/glsl-shaders), and [.slang(p)](https://github.com/libretro/slang-shaders) in particular, which are the most commonly used, but perhaps also [Quark shader folders](https://github.com/hizzlekizzle/quark-shaders) since they were used specifically for byuu/Near's SNES emulator. You can read a little more about the different shader/filter types on [Emulation General Wiki](https://emulation.gametechwiki.com/index.php/Shaders_and_filters) if you're so inclined.

The holdup was the different license types between the libretro shaders and not being sure if it was okay to add them to the repo, no? I still think the best way around this is to allow the user to incorporate them much like the MSU-1 feature.

**Reply by snesrev** [reply]
That's not really the only holdup. It's just not that straightforward for me to write the code for this, cause I never worked with shaders before.

**Comment by UltraHDR** [comment]
Dropped items don't disappear after x seconds.

Makes collecting items easier

**Comment by UltraHDR** [comment]
Use more than 2 bombs at once

**Reply by snesrev** [reply]
Added in master

**Comment by VictorXPDE** [comment]
Add the option to speed up the game by holding Space bar (or any other key you map it to) like the fast forward feature in emulators.

**Reply by snesrev** [reply]
Added in master

**Comment by UltraHDR** [comment]
Add Hero Mode https://zelda.fandom.com/wiki/Hero_Mode

Enemies deal double the damage and Hearts do not appear

**Comment by Covarr** [comment]
Idea: Feature for dungeon names appear when you enter a dungeon, as seen in the GBA port.
![image](https://user-images.githubusercontent.com/28900538/194578167-7e033878-9c34-403e-a728-30eecfcc0e51.png)

**Comment by ShadowOne333** [comment]
Possibly adding Redux as a feature would be neat :P
Here's the source code for it in case it's needed:
https://github.com/ShadowOne333/A-Link-to-the-Past-Redux

Also, another two suggestions I'd like to make:
1) Implement/unlock the Four Swords Dungeon from the GBA port upon completing the game once.
2)  port of Ancient Stone Tablets using the same engine as ALttP.

Just out of curiosity, is compiling the whole source back into an SNES ROM possibly with this?

**Reply by snesrev** [reply]
No there's no way to build the SNES rom from this.

**Reply by snesrev** [reply]
It would require a lot of changes to images/strings...

**Comment by JasonShearer** [comment]
How about an option to start a save with full hearts?

Edit: To clarify, I mean full health, not all heart pieces.

**Comment by JasonShearer** [comment]
Maybe an option to hide the output window?

**Comment by Kameo75** [comment]
Thank you for the effort!
Is it possible to hit it in ultrawide? can you do something to remove the black bars on the sides? thank you very much
![linktothepast](https://user-images.githubusercontent.com/122271973/211292340-93218cfd-b59d-4a48-83e9-eba62cf9a029.jpg)

**Reply by Kameo75** [reply]
I would also like to ask you how I can replace the textures.. a bit like in The legend of Zelda Nes (mesen remaster). If you can advise me.... thank you very much

**Comment by mdtauk** [comment]
How difficult would it be to implement the replacing of graphics and palettes, that break out of the limitations of the SNES image formats?

HD packs with more than we 256 colours for example?

**Comment by BenPaddon** [comment]
Optional HUD placement that mimics the placement of more recent Zelda games (HP/mana in top-left, current item(s) in top-right, rupee count in lower-left, arrow and bomb count aligned with those items when equipped)

**Comment by BenPaddon** [comment]
Optional spaced HUD to make better use of widescreen/ultrawide real estate.

**Comment by BenPaddon** [comment]
If one were to extract the English re-translation from the GBA version, would it be possibly to implement that as an optional alternate?

**Comment by Kallan007** [comment]
Can this be played on Steam Deck?

**Reply by DPS2004** [reply]
yes, if its compiled on the deck

**Reply by ghost** [reply]
Hi @Kallan007
I got it working on the deck quite easily. Although you can compile it on the deck, it's not necessary. I compiled it on my windows 11 pc, then transferred the parent folder to my deck using Winpinator/Warpinator detailed here: https://www.lifewire.com/connect-steam-deck-to-pc-6360959
Once you have added it to your steam library as a non-steam game, you'll need to go into the properties -> compatibility and choose the latest non-experimental version of proton. If I remember correctly that was the final step I had to do to get it working.
Good Luck!

**Reply by TheOldOne822** [reply]
Even that's not needed, it's in the Steam Decks built in "Discover" flatpak repro if you switch to desktop mode so you can just install it from there.

**Comment by Kryptortio** [comment]
Pause when the main window loses focus (and unpause when it gains focus).

**Comment by Guigou-hack** [comment]
Hello everyone ! I enjoy playing from my switch to zelda 3 in French again, thank you, however, after adding the dialogues in French, the menus remain in English. Would it be possible to add the hud_icons.png file in French in the extract in order to compile the whole game in French or in another language? THANKS !

**Comment by Guigou-hack** [comment]
![image](https://user-images.githubusercontent.com/57832367/227776326-ec39de1a-96c0-49e6-89a0-38993b7c5ae9.png)

**Comment by TheAnonymousCrusher** [comment]
Make having custom link sprites easier, heres an example:
<img width="128" height="448" alt="FoxPatch" src="https://github.com/user-attachments/assets/b0fe7302-26f0-493a-b2d2-428b3ad3e0e3" />
